### PR TITLE
Add API documentation for all new ADU functions

### PR DESCRIPTION
**Pull request by ewertons**
## Purpose
Add API documentation for all new ADU functions

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h [hunk 1]
```diff
@@ -169,13 +214,51 @@ AZ_NODISCARD az_result az_iot_adu_ota_get_properties_payload(
...
+ *                              previously parsed using 
+ *                              `az_iot_adu_ota_parse_service_properties`.
+ * @param[in] version           Version of the writable properties.
+ * @param[in] status            Azure Plug-and-Play status code for the 
+ *                              writable properties acknowledgement. 
+ * @param[in] payload           An #az_span pointing to the memory buffer where to
+ *                              write the resulting Azure Plug-and-Play properties.
+ * @param[out] out_payload      A pointer to hold the resulting #az_span slice of
+ *                              `payload`, fitted to the content generated by this
+ *                              function.
```
Comment (danewalton) [inline libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h:259]: just as a note, good wording here. I think it's much more descriptive of what's going on and maybe we can bring it to other parts of the embedded SDK.

File: libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h [hunk 2]
```diff
@@ -184,6 +267,20 @@ AZ_NODISCARD az_result az_iot_adu_ota_get_service_properties_response(
     az_span payload,
     az_span* out_payload);
 
+/**
+ * @brief Parses the json content from the ADU service update manifest into
+ *        a pre-defined structure.
+ * 
+ * @param[in] payload             ADU update manifest, as unescaped json content.
+ * @param[out] update_manifest    The structured where the parsed values of the
```
Comment (danewalton) [inline libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h:275]:
```suggestion
 * @param[out] update_manifest    The structure where the parsed values of the 
```

File: libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h [hunk 3]
```diff
@@ -291,24 +309,46 @@ AzureIoTResult_t AzureIoTADUClient_Init( AzureIoTADUClient_t * pxAduClient,
...
- * @param pxReader
- * @param ulPropertyVersion
- * @param pucWritablePropertyResponseBuffer
- * @param ulWritablePropertyResponseBufferSize
- * @param pulWritablePropertyResponseBufferLength
- * @return AzureIoTResult_t
+ * @brief Updates the ADU Agent Client with ADU service device update properties.
+ * @remark It must be called whenever writable properties are received containing
+ *         ADU service properties (verified with AzureIoTADUClient_IsADUComponent).
+ *         It effectivelly parses the properties (aka, the device update request)
```
Comment (danewalton) [inline libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h:330]:
```suggestion
 *         It effectively parses the properties (aka, the device update request)
```

File: libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h [hunk 4]
```diff
@@ -291,24 +309,46 @@ AzureIoTResult_t AzureIoTADUClient_Init( AzureIoTADUClient_t * pxAduClient,
...
+ * 
+ * @param[in] pxAduClient The #AzureIoTADUClient_t * to use for this call.
+ * @param[in] pxReader  A #AzureIoTJSONReader_t initialized with the ADU
+ *                      service writable properties json, set to the
+ *                      beginning of the json object that is the value
+ *                      of the ADU component.
+ * @param[in] ulPropertyVersion Version of the writable properties.
+ * @param[in] pucWritablePropertyResponseBuffer
+ *              An pointer to the memory buffer where to
+ *              write the resulting Azure Plug-and-Play properties acknoledgement
```
Comment (danewalton) [inline libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h:345]:
```suggestion
 *              write the resulting Azure Plug-and-Play properties acknowledgement
```

**Review by danewalton — APPROVED** [review]

File: libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h [hunk 5]
```diff
@@ -169,13 +214,51 @@ AZ_NODISCARD az_result az_iot_adu_ota_get_properties_payload(
...
+ */
 AZ_NODISCARD az_result az_iot_adu_ota_parse_service_properties(
     az_iot_hub_client const* iot_hub_client,
     az_json_reader* jr,
     az_span buffer,
     az_iot_adu_ota_update_request* update_request,
     az_span* buffer_remainder);
 
+/**
+ * @brief
```
[the remark this answers is not in the document]
Comment (ewertons) [reply libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_ota.h:246]:
```suggestion
 * @brief    Generates the payload necessary to respond to the service
             after receiving incoming properties.
```

**Pull request merged**